src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs

```diff
@@ -42,65 +41,44 @@ private void SaveTime(int hourOfDay, int minutes)
 				SaveInitialTime();
 			}
 		}
-		
-		private void ShowUsingTimePickerDialog()
-		{
-			_dialog = new TimePickerDialog(
-				      ContextHelper.Current,
-				      (sender, e) => SaveTime(e.HourOfDay, e.Minute),
-				      Time.Hours,
-				      Time.Minutes,
-				      ClockIdentifier == ClockIdentifiers.TwentyFourHour
-				      );
-
-			_dialog.DismissEvent += OnDismiss;
-			_dialog.Show();
-		}
 
-		private void ShowUsingAlertDialog()
+		private void ShowTimePicker(bool useSpinnerStyle)
 		{
-			//On Samsung devices ?= 6.0 the TimepickerDialog is not displayed properly in Landscape.
-			//Instead using this work around seems to work.
-			var timePicker = new Android.Widget.TimePicker(ContextHelper.Current);
+			var time = Time.RoundToInterval(MinuteIncrement);
+			var listener = new OnSetTimeListener((view, hourOfDay, minute) => SaveTime(hourOfDay, minute * MinuteIncrement));
 
-			//Hour and Minute properties are not supported on Android before 6.0
-			if (Build.VERSION.SdkInt >= BuildVersionCodes.M)
-			{
-				timePicker.Hour = Time.Hours;
-				timePicker.Minute = Time.Minutes;
-			}
-			else
-			{
-#pragma warning disable CS0618 // Type or member is obsolete
-				timePicker.CurrentHour = new Java.Lang.Integer(Time.Hours);
-				timePicker.CurrentMinute = new Java.Lang.Integer(Time.Minutes);
-#pragma warning restore CS0618 // Type or member is obsolete
-			}
-
-			timePicker.SetIs24HourView(new Java.Lang.Boolean(ClockIdentifier == ClockIdentifiers.TwentyFourHour));
-		
-			_dialog = new AlertDialog.Builder(ContextHelper.Current)
-				.SetPositiveButton(
-					Android.Resource.String.Ok,
-					(sender, eventArgs) =>
-						Time = new TimeSpan(Time.Days, timePicker.Hour, timePicker.Minute, Time.Seconds, Time.Milliseconds)
-				)
-				.SetNegativeButton(
-					Android.Resource.String.Cancel,
-					(sender, eventArgs) => { /*Nothing to do*/ }
-				)
-				.SetView(timePicker)
-				.Show();
+			_dialog = new UnoTimePickerDialog(
+					ContextHelper.Current,
+					useSpinnerStyle ? 3 : 0,
+					listener,
+					time.Hours,
+					time.Minutes,
+					ClockIdentifier == ClockIdentifiers.TwentyFourHour,
+					MinuteIncrement
+				);
 
 			_dialog.DismissEvent += OnDismiss;
+			_dialog.Show();
 		}
 
 		private void OnDismiss(object sender, EventArgs e)
 		{
 			Hide(canCancel: false);
 		}
 
-		partial void OnTimeChangedPartialNative(TimeSpan oldTime, TimeSpan newTime){}
+		partial void OnTimeChangedPartialNative(TimeSpan oldTime, TimeSpan newTime) { }
+
+		public class OnSetTimeListener : Java.Lang.Object, IOnTimeSetListener
```

Comment:
Instead of creating a new class, could you implement IOnTimeSetListener on TimePickerFlyout directly?  (and pass `this` to the dialog).

Reply:
I'm not entirely sure of the best practice. @jeromelaban

Reply:
class OnSetTimeListener need to implement Java.Lang.Object interface so i think it's better as internal class like now. 

public class OnSetTimeListener : Java.Lang.Object, IOnTimeSetListener